src/yb/tserver/pg_client_service.cc

```diff
@@ -724,14 +725,28 @@ class PgClientServiceImpl::Impl {
     return Status::OK();
   }
 
+  void GetRPCsWaitStates(tserver::PgActiveUniverseHistoryResponsePB* resp, rpc::Messenger* messenger) {
+    rpc::DumpRunningRpcsRequestPB dump_req;
+    rpc::DumpRunningRpcsResponsePB dump_resp;
+
+    dump_req.set_include_traces(false);
+    dump_req.set_get_wait_state(true);
+    dump_req.set_dump_timed_out(false);
+
+    WARN_NOT_OK(messenger->DumpRunningRpcs(dump_req, &dump_resp), "DumpRunningRpcs failed");
+    
+    for (auto conns : dump_resp.inbound_connections()) {
+      for (auto call : conns.calls_in_flight()) {
+        resp->add_wait_states()->CopyFrom(call.wait_state());
+      }
+    }
+  }
+
   Status ActiveUniverseHistory(
       const PgActiveUniverseHistoryRequestPB& req, PgActiveUniverseHistoryResponsePB* resp,
       rpc::RpcContext* context) {
-    auto tserver_wait_states = tablet_server_.ActiveUniverseHistory();
-
-    for (auto wait_state : tserver_wait_states) {
-      resp->add_wait_states()->CopyFrom(wait_state);
-    }
+    GetRPCsWaitStates(resp, tablet_server_.GetMessenger(util::MessengerType::kTserver));
```

Comment:
Should we store the TServer/CQL wait states separately in the ResponsePB?

Seems like we are just merging them into one list.

Reply:
We can separate them as tserver or cql by their wait states, so should we make still send them separately?

Reply:
We can do this as a follow-up. But, say we get 1000 events from Server, but 2000 events from CQL.

If we decide to keep only 500 events per service/server. They'd need to be sampled accordingly. So better to keep them separate.